src/ring_theory/polynomial/content.lean

```diff
@@ -0,0 +1,307 @@
+  apply dvd_sub (dvd.intro _ rfl) (dvd.intro _ rfl),
+end
+
+@[simp]
+theorem content_mul {p q : polynomial R} : (p * q).content = p.content * q.content :=
+begin
+  have h : ∀ (n : ℕ) (p q : polynomial R), (p * q).nat_degree ≤ n
+    → (p * q).content = p.content * q.content,
+  { clear p q,
+    intro n,
+    induction n with n hi,
+    { intros p q hdeg,
+      by_cases hp : p = 0, { simp [hp] },
+      by_cases hq : q = 0, { simp [hq] },
+      rw nat_degree_mul hp hq at hdeg,
+      cases nat.eq_zero_of_add_eq_zero (nat.eq_zero_of_le_zero hdeg) with pdeg qdeg,
+      rw [eq_C_of_nat_degree_eq_zero pdeg, content_C_mul, content_C] },
+    { intros p q hdeg,
+      by_cases hpdeg : p.nat_degree = 0,
+      { rw [eq_C_of_nat_degree_eq_zero hpdeg, content_C_mul, content_C] },
+      by_cases hqdeg : q.nat_degree = 0,
+      { rw [eq_C_of_nat_degree_eq_zero hqdeg, mul_comm, content_C_mul, content_C, mul_comm] },
+      by_cases hdeg' : (p * q).nat_degree = n.succ,
+      { rcases eq_C_mul_primitive p with ⟨cp, p1, rfl, p1_prim, p1_deg⟩,
+        rw [mul_assoc, content_C_mul, content_C_mul],
+        rcases eq_C_mul_primitive q with ⟨cq, q1, rfl, q1_prim, q1_deg⟩,
+        rw [mul_comm p1 _, mul_assoc, content_C_mul, content_C_mul, mul_comm q1 _],
+        suffices h : (p1 * q1).content = p1.content * q1.content,
+        { rw h, ring },
+        rw [nat_degree_mul, p1_deg, q1_deg] at hdeg',
+        rw p1_deg at hpdeg,
+        rw q1_deg at hqdeg,
+        iterate 2 { swap, intro h0,
+          apply nat.succ_ne_zero n,
+          rw [← hdeg', h0],
+          simp },
+        clear hdeg p1_deg q1_deg,
+        rw [p1_prim.content_eq_one, q1_prim.content_eq_one, mul_one,
+            content_eq_gcd_leading_coeff_content_erase_lead, leading_coeff_mul, ← normalize_gcd,
+            normalize_eq_one, is_unit_iff_dvd_one, gcd_comm],
+        transitivity,
+        apply gcd_mul_dvd_mul_gcd,
+        rw [← mul_one (1 : R)], apply mul_dvd_mul,
+        { rw [content_mul_aux, hi, q1_prim.content_eq_one, mul_one, gcd_comm,
+            ← content_eq_gcd_leading_coeff_content_erase_lead, p1_prim.content_eq_one],
+          cases erase_lead_nat_degree_lt_or_erase_lead_eq_zero p1,
+          { apply le_trans nat_degree_mul_le,
+            rw [← nat.lt_succ_iff, ← hdeg'],
+            apply nat.add_lt_add_right h },
+          { simp [h] } },
+        { rw [mul_comm, content_mul_aux, hi, p1_prim.content_eq_one, mul_one, gcd_comm,
+            ← content_eq_gcd_leading_coeff_content_erase_lead, q1_prim.content_eq_one],
+          cases erase_lead_nat_degree_lt_or_erase_lead_eq_zero q1,
+          { apply le_trans nat_degree_mul_le,
+            rw [← nat.lt_succ_iff, ← hdeg', add_comm],
+            apply nat.add_lt_add_left h },
+          { simp [h] } } },
+      { apply hi p q (nat.le_of_lt_succ (lt_of_le_of_ne hdeg hdeg')) } } },
+  apply h (p * q).nat_degree _ _  (le_refl _),
+end
```

Comment:
Is there any reasonable sublemma that you can extract from this?
If not... I guess we can just merge it.

Reply:
I haven't spun off a sublemma, but I did find a way to golf it down quite a bit.